libraries/standard/mqtt/src/mqtt.c

```diff
@@ -830,14 +954,33 @@ static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
                                                       pContext->transportInterface.networkContext,
                                                       pIncomingPacket );
 
-        /* Loop until there is data to read or if the timeout has not expired. */
-    } while( ( status == MQTTNoDataAvailable ) &&
-             ( calculateElapsedTime( getTimeStamp(), entryTimeMs ) < timeoutMs ) );
+        /* The loop times out based on 2 conditions.
+         * 1. If a non-NULL getTime function is provided by the application:
+         *    Loop times out based on the timeout calculated by getTime()
+         *    function.
+         * 2. If a non-NULL getTime function is not provided by the application:
+         *    Loop times out based on the maximum number of retries config
+         *    MQTT_CONNACK_RECEIVE_RETRY_COUNT. */
+        if( getTimeStamp != NULL )
+        {
+            /* Loop runs only once for 0 timeoutMs. */
+            breakFromLoop = ( bool ) ( calculateElapsedTime( getTimeStamp(), entryTimeMs ) >= timeoutMs );
```

Comment:
Is an explicit cast to `bool` type required? If the toolchain is C99 compliant, wouldn't it auto resolve to bool type? For C89 compilers, it would generate a 1 or 0 where `bool` would be aliased for `int8_t`

Reply:
The cast is not necessary for any compiler. This was entirely for MISRA, since it considers `1`s and `0`s obtained as a result of comparison/logical operators as "essentially boolean type" even in C89, where there is no boolean type. Note that this would still be an issue even if we didn't use the `bool` alias. However, this cast was another MISRA violation, so we ended up using a ternary `? true : false` at the end instead.